andrepereiradasilva/joomla-cms
# Html libraries performance improvements

Pull Request for Issue # .

### Summary of Changes

Code review and performance optimizations/improvements in the following clases:
- HTMLHelper
- JHtmlBehaviour
- JHtmlBootstrap
- JHtmlJQuery
- JHtmlFormBehaviour and chosen layout

More relevant notes:
- Added detectBrowser - false in all HTMLHelper js scripts loading so there is not searchs for 4 files (8 in debug mode) for file for each all possible relative paths - that's a lot a lstat filesystem calls (particulary when open_basedir config is enabled - see https://bugs.php.net/bug.php?id=52312).
The css files still have detect browser by default, but this should be removed as default.
- corrected a bug in HTMLHelper::extract key generation

- HTMLHelper::includeRelativeFiles was not changed since needs more work

### Testing Instructions

Mainly code review and navigate around the site/admin to check if everything is fine.

### Documentation Changes Required

None.
